src/main/java/com/greghaskins/spectrum/Spectrum.java

```diff
@@ -188,7 +205,8 @@ public static Configuration configure() {
    * @param block {@link com.greghaskins.spectrum.Block} to run once before each spec
    */
   public static void beforeEach(final com.greghaskins.spectrum.Block block) {
-    getCurrentSuiteBeingDeclared().beforeEach(block);
+    addHook(new HookContext(before(block), getDepth(),
+        HookContext.AppliesTo.ATOMIC_ONLY, HookContext.Precedence.LOCAL));
```

Comment:
all hookcontexts created by `Spectrum` which finds it is to provide the depth.

The Precedence names and application are arbitrary and "whatever I had to do to get the tests to work"